{"uid":"b9550b087027efc3","name":"test_relation_with_self_signed_certificates_operator","fullName":"tests.integration.test_tls#test_relation_with_self_signed_certificates_operator","historyId":"92f6dc227aca8c1ff6658d63b5f1e708","time":{"start":1706672305392,"stop":1706672307401,"duration":2009},"description":"Test the relation with the Self Signed Certificates operator.","descriptionHtml":"<p>Test the relation with the Self Signed Certificates operator.</p>\n","status":"broken","statusMessage":"juju.errors.JujuAPIError: cannot add relation \"postgresql:certificates self-signed-certificates:certificates\": establishing a new relation for postgresql:certificates would exceed its maximum relation limit of 1","statusTrace":"Traceback (most recent call last):\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 341, in from_call\n    result: Optional[TResult] = func()\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 262, in <lambda>\n    lambda: ihook(item=item, **kwds), when=when, reraise=reraise\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_hooks.py\", line 493, in __call__\n    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_manager.py\", line 115, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 152, in _multicall\n    return outcome.get_result()\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_result.py\", line 114, in get_result\n    raise exc.with_traceback(exc.__traceback__)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 77, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 177, in pytest_runtest_call\n    raise e\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 169, in pytest_runtest_call\n    item.runtest()\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/python.py\", line 1792, in runtest\n    self.ihook.pytest_pyfunc_call(pyfuncitem=self)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_hooks.py\", line 493, in __call__\n    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_manager.py\", line 115, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 152, in _multicall\n    return outcome.get_result()\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_result.py\", line 114, in get_result\n    raise exc.with_traceback(exc.__traceback__)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 77, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/python.py\", line 194, in pytest_pyfunc_call\n    result = testfunction(**testargs)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pytest_asyncio/plugin.py\", line 532, in inner\n    _loop.run_until_complete(task)\n  File \"/usr/lib/python3.10/asyncio/base_events.py\", line 649, in run_until_complete\n    return future.result()\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/tests/integration/test_tls.py\", line 219, in test_relation_with_self_signed_certificates_operator\n    await ops_test.model.relate(DATABASE_APP_NAME, SELF_SIGNED_CERTIFICATES_APP_NAME)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/juju/model.py\", line 1554, in relate\n    return await self.integrate(relation1, relation2)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/juju/model.py\", line 1532, in integrate\n    result = await app_facade.AddRelation(endpoints=endpoints, via_cidrs=None)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/juju/client/facade.py\", line 486, in wrapper\n    reply = await f(*args, **kwargs)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/juju/client/_client17.py\", line 890, in AddRelation\n    reply = await self.rpc(msg)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/juju/client/facade.py\", line 659, in rpc\n    result = await self.connection.rpc(msg, encoder=TypeEncoder)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/juju/client/connection.py\", line 679, in rpc\n    raise errors.JujuAPIError(result)\njuju.errors.JujuAPIError: cannot add relation \"postgresql:certificates self-signed-certificates:certificates\": establishing a new relation for postgresql:certificates would exceed its maximum relation limit of 1","flaky":false,"newFailed":false,"newBroken":true,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"event_loop","time":{"start":1706670781846,"stop":1706670781847,"duration":1},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"tmp_path_factory","time":{"start":1706670781847,"stop":1706670781847,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"ops_test","time":{"start":1706670781847,"stop":1706670782159,"duration":312},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"abort_on_fail","time":{"start":1706672305391,"stop":1706672305391,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"testStage":{"description":"Test the relation with the Self Signed Certificates operator.","status":"broken","statusMessage":"juju.errors.JujuAPIError: cannot add relation \"postgresql:certificates self-signed-certificates:certificates\": establishing a new relation for postgresql:certificates would exceed its maximum relation limit of 1","statusTrace":"Traceback (most recent call last):\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 341, in from_call\n    result: Optional[TResult] = func()\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 262, in <lambda>\n    lambda: ihook(item=item, **kwds), when=when, reraise=reraise\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_hooks.py\", line 493, in __call__\n    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_manager.py\", line 115, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 152, in _multicall\n    return outcome.get_result()\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_result.py\", line 114, in get_result\n    raise exc.with_traceback(exc.__traceback__)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 77, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 177, in pytest_runtest_call\n    raise e\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 169, in pytest_runtest_call\n    item.runtest()\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/python.py\", line 1792, in runtest\n    self.ihook.pytest_pyfunc_call(pyfuncitem=self)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_hooks.py\", line 493, in __call__\n    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_manager.py\", line 115, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 152, in _multicall\n    return outcome.get_result()\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_result.py\", line 114, in get_result\n    raise exc.with_traceback(exc.__traceback__)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 77, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/python.py\", line 194, in pytest_pyfunc_call\n    result = testfunction(**testargs)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pytest_asyncio/plugin.py\", line 532, in inner\n    _loop.run_until_complete(task)\n  File \"/usr/lib/python3.10/asyncio/base_events.py\", line 649, in run_until_complete\n    return future.result()\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/tests/integration/test_tls.py\", line 219, in test_relation_with_self_signed_certificates_operator\n    await ops_test.model.relate(DATABASE_APP_NAME, SELF_SIGNED_CERTIFICATES_APP_NAME)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/juju/model.py\", line 1554, in relate\n    return await self.integrate(relation1, relation2)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/juju/model.py\", line 1532, in integrate\n    result = await app_facade.AddRelation(endpoints=endpoints, via_cidrs=None)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/juju/client/facade.py\", line 486, in wrapper\n    reply = await f(*args, **kwargs)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/juju/client/_client17.py\", line 890, in AddRelation\n    reply = await self.rpc(msg)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/juju/client/facade.py\", line 659, in rpc\n    result = await self.connection.rpc(msg, encoder=TypeEncoder)\n  File \"/home/ubuntu/github-runner/_work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/juju/client/connection.py\", line 679, in rpc\n    raise errors.JujuAPIError(result)\njuju.errors.JujuAPIError: cannot add relation \"postgresql:certificates self-signed-certificates:certificates\": establishing a new relation for postgresql:certificates would exceed its maximum relation limit of 1","steps":[],"attachments":[{"uid":"5dd5f640e2df351","name":"log","source":"5dd5f640e2df351.txt","type":"text/plain","size":4617}],"parameters":[],"stepsCount":0,"shouldDisplayMessage":true,"attachmentsCount":1,"hasContent":true,"attachmentStep":false},"afterStages":[{"name":"abort_on_fail::0","time":{"start":1706672307406,"stop":1706672307406,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"ops_test::finalizer","time":{"start":1706672307407,"stop":1706672308060,"duration":653},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"event_loop::2","time":{"start":1706672308060,"stop":1706672308061,"duration":1},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"event_loop::_close_event_loop","time":{"start":1706672308061,"stop":1706672308061,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"event_loop::_provide_clean_event_loop","time":{"start":1706672308061,"stop":1706672308061,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"labels":[{"name":"tag","value":"group(1)"},{"name":"tag","value":"@pytest.mark.skipif(False, reason='Requires juju 3')"},{"name":"tag","value":"asyncio"},{"name":"parentSuite","value":"tests.integration"},{"name":"suite","value":"test_tls"},{"name":"host","value":"large-224-f73406e3a949408aa81f672f"},{"name":"thread","value":"7011-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.integration.test_tls"},{"name":"resultFormat","value":"allure2"}],"parameters":[],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[],"categories":[{"name":"Test defects","matchedStatuses":[],"flaky":false}],"history":{"statistic":{"failed":0,"broken":1,"skipped":0,"passed":3,"unknown":0,"total":4},"items":[{"uid":"af63341d3d8c1bf0","reportUrl":"../1928//#testresult/af63341d3d8c1bf0","status":"passed","time":{"start":1706579608447,"stop":1706579989223,"duration":380776}},{"uid":"4e5d52e7446e8025","reportUrl":"../1919//#testresult/4e5d52e7446e8025","status":"passed","time":{"start":1706492983530,"stop":1706493296284,"duration":312754}},{"uid":"57189e15e97a1515","reportUrl":"../1917//#testresult/57189e15e97a1515","status":"passed","time":{"start":1706406855876,"stop":1706407207374,"duration":351498}}]},"tags":["asyncio","@pytest.mark.skipif(False, reason='Requires juju 3')","group(1)"]},"source":"b9550b087027efc3.json","parameterValues":[]}